### PR TITLE
python312Packages.{matchpy,uarray}: unbreak

### PR DESCRIPTION
**Pull request by pbsds**
## Description of changes

fixes https://hydra.nixos.org/build/254884740
closes https://github.com/NixOS/nixpkgs/pull/304981

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/uarray/default.nix [hunk 1]
```diff
@@ -12,25 +14,17 @@
...
-  format = "setuptools";
+  version = "0.8.8";
+  pyproject = true;
 
   src = fetchFromGitHub {
     owner = "Quansight-Labs";
     repo = pname;
     rev = version;
-    sha256 = "1x2jp7w2wmn2awyv05xs0frpq0fa0rprwcxyg72wgiss0bnzxnhm";
+    sha256 = "sha256-wTKqOw64b+/kdZpSYLwCJATOuo807BWCtVHB4pH58fY=";
```
Comment (kirillrdy) [inline pkgs/development/python-modules/uarray/default.nix:24]:
```suggestion
    hash = "sha256-wTKqOw64b+/kdZpSYLwCJATOuo807BWCtVHB4pH58fY=";
```

i know its annoying, but bit by bit

Comment (pbsds) [reply pkgs/development/python-modules/uarray/default.nix]: preferably we'd do this in treewides rather than block a bunch of PRs

**Review by kirillrdy — APPROVED** [review]
Result of `nixpkgs-review pr 305016` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.loopy</li>
    <li>python311Packages.loopy.dist</li>
    <li>python312Packages.loopy</li>
    <li>python312Packages.loopy.dist</li>
  </ul>
</details>
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>python311Packages.matchpy</li>
    <li>python311Packages.matchpy.dist</li>
    <li>python311Packages.pymbolic</li>
    <li>python311Packages.pymbolic.dist</li>
    <li>python311Packages.uarray</li>
    <li>python311Packages.uarray.dist</li>
    <li>python312Packages.matchpy</li>
    <li>python312Packages.matchpy.dist</li>
    <li>python312Packages.pymbolic</li>
    <li>python312Packages.pymbolic.dist</li>
    <li>python312Packages.uarray</li>
    <li>python312Packages.uarray.dist</li>
  </ul>
</details>

**Comment by pbsds** [comment]
loopy failure expected https://hydra.nixos.org/build/256259504, and its issue seem rather involved

**Pull request merged**